builtin/mainmenu/modmgr.lua

```diff
@@ -440,30 +440,29 @@ function modmgr.preparemodlist(data)
 
 	modmgr.mods_by_name = {}
 	local mods_active_by_name = {}
-	for key,value in pairs(worldfile:to_table()) do
-		if key:sub(1, 9) == "load_mod_" then
-			key = key:sub(10)
-			local element = nil
-			for i=1,#retval,1 do
-				if retval[i].name == key and
-						not retval[i].is_modpack then
-					element = retval[i]
-					modmgr.mods_by_name[element.name] = modmgr.mods_by_name[element.name] or {}
-					table.insert(modmgr.mods_by_name[element.name], element)
-					local is_true = core.is_yes(value)
-					if is_true and element.typ == "global_mod" then
-						mods_active_by_name[element.name] = mods_active_by_name[element.name] or {}
-						table.insert(mods_active_by_name[element.name], element)
-					end
-					element.enabled = is_true or value == get_modpack_path(element)
+	local missing_configured_mods = worldfile:to_table()
+	for i, mod in ipairs(retval) do
+		if not mod.is_modpack then
+			modmgr.mods_by_name[mod.name] = modmgr.mods_by_name[mod.name] or {}
+			table.insert(modmgr.mods_by_name[mod.name], mod)
+			local key = "load_mod_" .. mod.name
+			local value = worldfile:get(key)
+			missing_configured_mods[key] = nil
+			if value then
+				local is_true = core.is_yes(value)
+				if is_true and mod.typ == "global_mod" then
+					mods_active_by_name[mod.name] = mods_active_by_name[nid.name] or {}
+					table.insert(mods_active_by_name[mod.name], mod)
 				end
-			end
-			if element == nil then
-				core.log("info", "Mod: " .. key .. " " .. dump(value) .. " but not found")
+				mod.enabled = is_true or value == get_modpack_path(mod)
 			end
 		end
 	end
 
+	for key, value in pairs(missing_configured_mods) do
+		core.log("info", "Mod: " .. key .. " " .. dump(value) .. " but not found")
```

Comment:
The `info` log level does not appear in the terminal but the text is written to the debug.txt file.
```
2018-01-15 19:20:05: INFO[Main]: Mod: backend "sqlite3" but not found
2018-01-15 19:20:05: INFO[Main]: Mod: server_announce "false" but not found
2018-01-15 19:20:05: INFO[Main]: Mod: creative_mode "false" but not found
2018-01-15 19:20:05: INFO[Main]: Mod: enable_damage "false" but not found
2018-01-15 19:20:05: INFO[Main]: Mod: gameid "minetest" but not found
2018-01-15 19:20:05: INFO[Main]: Mod: player_backend "sqlite3" but not found
```
It should check against the `load_mod_* ~= false` pattern here.
Also: `"Mod: " .. key .. " " .. dump(value) .. " but was not found"`

Reply:
I'd say the wording is still cryptic. How about `Config says that mod "mobs" should be loaded from "mods"' but the mod was not found there`?

Reply:
either that or `The mod "mobs" located in the directory "mods" cannot be found.`.
That's similar to what Minetest logs as `error` when joining the world: `The following mods could not be found: "mobs"`

PS, FYI: I usually upvote on review comments as a feedback that it has been read.

Reply:
Thanks. My wording hints that the value of the setting is the place where it should be found, which I'd prefer, and also it's not a directory properly speaking, because it uses dots, so I'll opt for my wording.

The reason for keeping that message was to not remove existing functionality. Not sure if I went too far.